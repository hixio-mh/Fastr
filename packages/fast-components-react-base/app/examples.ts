import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import DividerExamples from "../src/divider/examples.data";
export { DividerExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export {LabelExamples};

import TextField from "../src/text-field/examples.data";
export { TextField };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
