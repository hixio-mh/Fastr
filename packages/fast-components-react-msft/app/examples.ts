import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import DividerExample from "../src/divider/examples.data";
export { DividerExample };

import FlipperExample from "../src/flipper/examples.data";
export { FlipperExample };

import HeadingExample from "../src/heading/examples.data";
export { HeadingExample };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };
