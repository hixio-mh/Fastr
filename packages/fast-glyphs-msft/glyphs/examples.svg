<svg width="16" height="16" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="Canvas">
<g id="Group">
<g id="Vector">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.3 5.8L 2.3 3.9L 0.6 4.8L -3.8147e-07 3.8L 1.7 2.9L 0.1 2L 0.7 1L 2.2 1.9L 2.2 2.38419e-08L 3.3 2.38419e-08L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(10.3999 1.09998)"/>
</g>
<g id="Vector_2">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.2 5.8L 2.2 3.9L 0.6 4.8L 4.76837e-08 3.8L 1.6 2.9L 4.76837e-08 2L 0.6 1L 2.2 1.9L 2.2 -1.90735e-07L 3.3 -1.90735e-07L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(1.19995 10.2)"/>
</g>
<g id="Vector_3">
<path d="M 0 8L 0 0L 8 0L 8 8L 0 8ZM 1 1L 1 7L 7 7L 7 1L 1 1ZM 14.4 13.1L 16 14L 15.4 15L 13.8 14.1L 13.8 16L 12.7 16L 12.7 14.1L 11 15.1L 10.4 14.1L 12 13.2L 10.4 12.3L 11 11.3L 12.6 12.2L 12.6 10.3L 13.7 10.3L 13.7 12.2L 15.3 11.3L 15.9 12.3L 14.4 13.1Z"/>
</g>
</g>
</g>
</svg>