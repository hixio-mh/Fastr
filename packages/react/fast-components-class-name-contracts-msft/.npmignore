# Source files
src/

# Configs
tsconfig.json