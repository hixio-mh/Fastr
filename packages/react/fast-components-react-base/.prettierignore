coverage/*
dist/*
www/*
src/**/.tmp/