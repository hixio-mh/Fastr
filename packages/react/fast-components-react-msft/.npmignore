# Tests
__test__/
*.spec.*
*.test.*

# Source files
app/
www/
build/
coverage/
src/
assets/

# config
babel.config.js
tsconfig.json
webpack.config.js