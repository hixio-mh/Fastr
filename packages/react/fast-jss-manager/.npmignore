# Source files
src/

# config
tsconfig.json