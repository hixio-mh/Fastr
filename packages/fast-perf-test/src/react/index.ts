import React from "react";
import ReactDOM from "react-dom";
import { Checkbox, TextField } from "@microsoft/fast-components-react-msft";

const root = document.createElement("div");
document.body.appendChild(root);

function getQueryVariable(variable) {
    var query = window.location.search.substring(1);
    var vars = query.split("&");
    for (var i = 0; i < vars.length; i++) {
        var pair = vars[i].split("=");
        if (decodeURIComponent(pair[0]) == variable) {
            return decodeURIComponent(pair[1]);
        }
    }
}

const count = getQueryVariable("count");

function renderItems() {
    return React.createElement(
        React.Fragment,
        {},
        new Array(typeof count === "string" ? (count as any) * 1 : 2000)
            .fill(0)
            .map((item, index) => React.createElement(TextField))
    );
}

function render(): void {
    ReactDOM.render(React.createElement(renderItems), root);
}

render();
