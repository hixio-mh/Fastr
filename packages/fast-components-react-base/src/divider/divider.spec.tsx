import * as React from "react";
import * as ShallowRenderer from "react-test-renderer/shallow";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import {
    generateSnapshots,
    SnapshotTestSuite,
} from "@microsoft/fast-jest-snapshots-react";
import Divider, {
    DividerClassNameContract,
    DividerHandledProps,
    DividerManagedClasses,
    DividerProps,
    DividerRoles,
    DividerUnhandledProps,
} from "./divider";

describe("divider snapshot", (): void => {
    generateSnapshots(examples as SnapshotTestSuite<DividerProps>);
});

configure({ adapter: new Adapter() });

describe("divider", (): void => {
    const managedClasses: DividerClassNameContract = {
        divider: "divider-class",
    };

    test("should have a displayName that matches the component name", () => {
        expect((Divider as any).name).toBe(Divider.displayName);
    });

    test("should not throw if managedClasses are not provided", () => {
        expect(() => {
            shallow(<Divider />);
        }).not.toThrow();
    });

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: DividerHandledProps = {
            managedClasses,
        };
        const unhandledProps: DividerUnhandledProps = {
            "aria-hidden": true,
        };
        const props: DividerProps = { ...handledProps, ...unhandledProps };
        const rendered: any = shallow(<Divider {...props} />);

        expect(rendered.prop("aria-hidden")).not.toBe(undefined);
        expect(rendered.prop("aria-hidden")).toEqual(true);
    });

    test("should pass `role` prop when value is `presentation`", () => {
        const rendered: any = shallow(
            <Divider managedClasses={managedClasses} role={DividerRoles.presentation} />
        );

        expect(rendered.prop("role")).toEqual(DividerRoles.presentation);
    });

    test("should NOT have a role attribute when value is `separator`", () => {
        const rendered: any = shallow(
            <Divider managedClasses={managedClasses} role={DividerRoles.separator} />
        );

        expect(rendered.prop("role")).not.toEqual(DividerRoles.separator);
        expect(rendered.prop("role")).toBe(undefined);
    });
});
