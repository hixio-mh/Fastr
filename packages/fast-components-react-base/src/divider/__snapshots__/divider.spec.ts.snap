// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`divider Divider: 0 1`] = `
<hr
  className="divider"
/>
`;
